src/datasources/cache/redis.cache.service.spec.ts

```diff
@@ -77,147 +69,100 @@ describe('RedisCacheService', () => {
+      await redisClient.hGet(`invalidationTimeMs:${cacheDir.key}`, ''),
+    );
+    const invalidationTimeTtl = await redisClient.ttl(
+      `invalidationTimeMs:${cacheDir.key}`,
     );
-    expect(redisClientTypeMock.expire).toHaveBeenCalledTimes(1);
-    expect(redisClientTypeMock.expire).toHaveBeenCalledWith(
-      `invalidationTimeMs:${key}`,
+    expect(storedValue).toBeNull();
+    expect(invalidationTime).toBeGreaterThanOrEqual(startTime);
+    expect(invalidationTimeTtl).toBeGreaterThan(0);
+    expect(invalidationTimeTtl).toBeLessThanOrEqual(
       defaultExpirationTimeInSeconds,
     );
-    expect(redisClientTypeMock.hDel).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.quit).toHaveBeenCalledTimes(0);
-    jest.useRealTimers();
   });
 
-  it('Deleting keys by pattern calls scan and unlink', async () => {
+  it('Deleting keys by pattern deletes matching keys only', async () => {
+    const pattern = faker.string.alphanumeric();
     const matches = [
-      faker.string.alphanumeric(),
-      faker.string.alphanumeric(),
-      faker.string.alphanumeric(),
+      `${pattern}${faker.string.alphanumeric()}`,
+      `${pattern}${faker.string.alphanumeric()}`,
+      `${pattern}${faker.string.alphanumeric()}`,
     ];
-    redisClientTypeMock.scanIterator = jest.fn().mockReturnValue(matches);
+    const anotherPattern = faker.string.alphanumeric();
+    const value = fakeJson();
+    await redisClient.hSet(anotherPattern, '', value);
+    for (const key of matches) {
+      await redisClient.hSet(key, '', value);
+    }
 
-    await redisCacheService.deleteByKeyPattern(faker.string.alphanumeric());
+    await redisCacheService.deleteByKeyPattern(`${pattern}*`);
 
-    expect(redisClientTypeMock.scanIterator).toHaveBeenCalledTimes(1);
-    expect(redisClientTypeMock.unlink).toHaveBeenCalledTimes(matches.length);
-    expect(redisClientTypeMock.hGet).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.hSet).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.hDel).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.quit).toHaveBeenCalledTimes(0);
+    for (const key of matches) {
+      expect(await redisClient.hGet(key, '')).toBeNull();
+    }
+    expect(await redisClient.hGet(anotherPattern, '')).toEqual(value);
   });
 
   it('When Module gets destroyed, redis connection is closed', async () => {
     await redisCacheService.onModuleDestroy();
 
-    expect(redisClientTypeMock.hGet).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.hSet).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.hDel).toHaveBeenCalledTimes(0);
-    expect(redisClientTypeMock.quit).toHaveBeenCalledTimes(1);
+    // Connection is closed, this is expected to throw an error
+    await expect(redisCacheService.ping()).rejects.toThrow();
+    // Connection is reopened after this test execution
+    redisClient = await redisClientFactory();
+  });
+
+  afterAll(async () => {
+    await redisClient.flushAll();
```

Comment:
This can affect other tests. For now the prefix avoid conflicts in the key-space but if we trigger a `FLUSHALL` on the instance then we might affect the state of the tests that depend on the same instance.